packages/flutter/lib/src/rendering/paragraph.dart

```diff
@@ -305,6 +305,9 @@ class RenderParagraph extends RenderBox with ContainerRenderObjectMixin<RenderBo
   static final String _placeholderCharacter = String.fromCharCode(PlaceholderSpan.placeholderCodeUnit);
   final TextPainter _textPainter;
 
+  /// The painter that controls how the text is painted.
+  TextPainter get textPainter => _textPainter;
```

Comment:
This is an implementation detail of the RenderParagraph and the RenderParagraph makes certain assumptions about the state it is in. We shouldn't make this public and let anyone mess with this.

cc @justinmc @LongCatIsLooong

Reply:
Hey @goderbauer I'm only interested in reading the data, exposing line metrics which I'm assuming is immutable is enough.

Reply:
Could this be done via a `LineMetrics` getter then instead of exposing the painter?

Reply:
Totally, ill update the PR